File: 538. Convert BST to Greater Tree/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private int sum;
    
    public TreeNode convertBST(TreeNode root) {
        ConvertTree(root);
            
        return root;
    }
     
    private void ConvertTree(TreeNode root) {
        if(root == null)
            return;
        
        ConvertTree(root.right);
        
        root.val += sum;
        sum = root.val;
        
        ConvertTree(root.left);
    }
}

File: 538. Convert BST to Greater Tree/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
private :
    int sum = 0;
    void helper(TreeNode* node) {
        if(node->right)
            helper(node->right);
        
        sum += node->val;
        node->val = sum;
        
        if(node->left)
            helper(node->left);
    }
    
public:
    TreeNode* convertBST(TreeNode* root) {
        if (!root) 
            return root;
        
        helper(root);
        return root;
    }
};